lib/capybara/driver/node.rb

```diff
@@ -138,6 +156,27 @@ def inspect
       def ==(other)
         eql?(other) || (other.respond_to?(:native) && native == other.native)
       end
+
+      protected
+
+      def normalize_spacing(text)
+        text
+          .delete(REMOVED_CHARACTERS)
+          .tr(SQUEEZED_SPACES, ' ')
+          .squeeze(' ')
+          .sub(LEADING_SPACES, '')
+          .sub(TRAILING_SPACES, '')
+          .tr(NON_BREAKING_SPACE, ' ')
+      end
+
+      def normalize_visible_spacing(text)
+        text
+          .squeeze(' ')
+          .gsub(EMPTY_LINES, "\n")
+          .sub(LEADING_SPACES, '')
+          .sub(TRAILING_SPACES, '')
+          .tr(NON_BREAKING_SPACE, ' ')
```

Comment:
I wonder if using `!` versions on a `text.dup` would be any faster 🤔  It would be only 1 object created vs 5.

Reply:
May be worth trying out.